lib/truffle/java/array_java_proxy.rb

```diff
@@ -6,19 +6,65 @@
 # GNU General Public License version 2
 # GNU Lesser General Public License version 2.1
 
-class ArrayJavaProxy < java.lang.Object
-  include Enumerable
-  
-  def each
-    (0...size).each do |i|
-      yield(self[i])
-    end
-    self
+begin
+  Thread.current[:MAKE_PROXY] = true
+  class ArrayJavaProxy < java.lang.Object
+    include Enumerable
+
+    def each
+      (0...size).each do |i|
+        yield(self[i])
```

Comment:
I wouldn't normally put parens around yield's argument like this - it's a keyword.